sdk/core/Azure.Core/src/Shared/MessagingClientDiagnostics.cs

```diff
@@ -94,15 +94,15 @@ public MessagingClientDiagnostics(string clientNamespace, string? resourceProvid
         /// <param name="traceparent">The trace parent of the message.</param>
         /// <param name="tracestate">The trace state of the message.</param>
         /// <returns><c>true</c> if the message properties contained the diagnostic id; otherwise, <c>false</c>.</returns>
-        public static bool TryExtractTraceContext(IReadOnlyDictionary<string, object> properties, out string? traceparent, out string? tracestate)
+        public static bool TryExtractTraceContext(IReadOnlyDictionary<string, object?> properties, out string? traceparent, out string? tracestate)
```

Comment:
These updates were necessary to support targeting net6.0 in consuming libraries.